tests/test_diffeq/test_odefiltsmooth/test_odefiltsmooth.py

```diff
@@ -1,507 +1,108 @@
-
-    def test_filter_ivp_h_mat52_ukf(self):
-        """UKF requires some evaluation-variance to have a positive definite innovation
-        matrix, apparently."""
-        probsolve_ivp(
-            self.ivp, step=self.step, evlvar=0.01, which_prior="matern52", method="uks"
-        )
-
-    def test_filter_ivp_h_mat72_kf(self):
-        probsolve_ivp(self.ivp, step=self.step, which_prior="matern72", method="eks0")
-
-
-class TestPreconditioning(unittest.TestCase):
-    """Solver with high order and small stepsize should work up to a point where
-    step**order is below machine precision."""
-
-    def setUp(self):
-        initdist = Constant(20 * np.ones(2))
-        self.ivp = ode.lotkavolterra([0.0, 1e-4], initdist)
-        self.step = 1e-5
-        self.prior = "ibm3"
 
-    def test_small_step_feasible(self):
-        """With the 'old' preconditioner, this is impossible because step**(2*order + 1)
-        is too small.
 
-        With the 'new' preconditioner, the smallest value that appears
-        in the solver code is step**order
-        """
-        probsolve_ivp(self.ivp, step=self.step, which_prior=self.prior, method="eks0")
+@pytest.fixture
+def ivp():
+    initrv = Constant(20.0 * np.ones(2))
+    return ode.lotkavolterra([0.0, 0.25], initrv)
+
+
+@pytest.mark.parametrize("method", ["EK0", "EK1"])
+@pytest.mark.parametrize(
+    "which_prior",
+    [
+        "IBM1",
+        "IBM2",
+        "IBM3",
+        "IBM5",
+        "IBM8",
+        "IBM11",
+        "IOUP1",
+        "IOUP2",
+        "IOUP3",
+        "IOUP4",
+        "MAT32",
+        "MAT52",
+        "MAT72",
+        "MAT92",
+    ],
+)
+@pytest.mark.parametrize("dense_output", [True, False])
+@pytest.mark.parametrize("step", [0.01, None])
+def test_adaptive_solver_successful(ivp, method, which_prior, dense_output, step):
+    """The solver terminates successfully for all sorts of parametrizations."""
+    f = ivp.rhs
+    df = ivp.jacobian
+    t0, tmax = ivp.timespan
+    y0 = ivp.initrv.mean
+
+    lengthscale = 10.0
+    driftspeed = 1.0
+
+    sol = probsolve_ivp(
+        f,
+        t0,
+        tmax,
+        y0,
+        df=df,
+        atol=1e-1,
+        rtol=1e-1,
+        which_prior=which_prior,
+        method=method,
+        dense_output=dense_output,
+        step=step,
+        driftspeed=driftspeed,
+        lengthscale=lengthscale,
+    )
+    # Successful return value as documented
+    assert isinstance(sol, KalmanODESolution)
+
+    # Adaptive steps are not evenly distributed
+    step_diff = np.diff(sol.t)
+    step_ratio = np.amin(step_diff) / np.amax(step_diff)
+    assert step_ratio < 0.5
+
+
+def test_wrong_prior_raises_error(ivp):
+    """Priors that are not in the list raise errors."""
+    f = ivp.rhs
+    t0, tmax = ivp.timespan
+    y0 = ivp.initrv.mean
+
+    # Anything that is no {IBM, IOUP, MAT} + Number is wrong
+    # (the Matern number must end on a 2).
+    for which_prior in ["IBM_5", "IOUPX5", "MAT112Y", "MAT33"]:
```

Comment:
any other corner cases that may be checked here? These seemed like the most obvious ones (all of which are naturally captured by the RE -- thanks!)